include/APIClient/GPU.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <restclient-cpp/restclient.h>
+#include <restclient-cpp/connection.h>
+
+#include <APIClient/Common.hpp>
+#include <APIClient/Router.hpp>
+
+namespace APIClient {
+
+class GPU {
+
+    using ResponseCode = int;
+    using JSONResponse = std::tuple<ResponseCode, nlohmann::json>;
+
+public:
+    GPU() = default;
+
+    GPU(const GPU&) = delete;
+
+    GPU& operator=(const GPU&) = delete;
+    GPU& operator=(const GPU&&) = delete;
+
+    ~GPU() = default;
+
+    JSONResponse put(nlohmann::json vulkanInfosJson);
```

Comment:
You need to take a `const nlohmann::json&`.
And you can probably rename the variable (the same in the .cpp) `gpuInfo`